src/Cardano/Wallet/DB/MVar.hs

```diff
@@ -33,36 +39,97 @@ import Data.Map.Strict
 
 import qualified Data.Map.Strict as Map
 
+data Database s = Database
+    { wallet :: Wallet s
+    , metadata :: WalletMetadata
+    , txHistory :: Map (Hash "Tx") (Tx, TxMeta)
```

Comment:
cannot understand why we need `TxMeta` (not related to this PR)

Reply:
We _need_ them because our users need them ;)
Those pieces of information (direction, timestamp, total amount etc..) are very valuable to API consumers.